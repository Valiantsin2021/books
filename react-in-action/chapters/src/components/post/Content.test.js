import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Content from "../../../src/components/post/Content";

// ch[9.1] test with shallow rendering
describe("<Content/>", () => {
  describe("render methods", () => {
    const mockPost = {
      content: "I am learning to test React components"
    };
    test("should render correctly with Enzyme's shallow method", () => {
      // use Enzyme's shallow method to render the component
      const wrapper = shallow(<Content post={mockPost} />);
      expect(wrapper.find("p").length).toBe(1);
      expect(wrapper.find("p.content").length).toBe(1);
      expect(wrapper.find(".content").text()).toBe(mockPost.content);
      expect(wrapper.find("p").text()).toBe(mockPost.content);
    });
    test("use renderer to create snapshot, and Jest to test it", () => {
      // use renderer to create snapshot, and Jest to test it
      const component = renderer.create(<Content post={mockPost} />);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
