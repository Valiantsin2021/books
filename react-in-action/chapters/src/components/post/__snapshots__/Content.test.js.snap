// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Content/> render methods use renderer to create snapshot, and Jest to test it 1`] = `
<p
  className="content"
>
  I am learning to test React components
</p>
`;
